src/test/java/br/com/sbk/sbking/core/HandEvaluationsTest.java

```diff
@@ -0,0 +1,201 @@
+
+    Hand firstHand = this.createMockedHandWithDistribution(this.createSuitDistribution(1, 1, 0, 1));
+
+    int numberOfHearts = 4;
+    Hand secondHand = this.createMockedHandWithDistribution(this.createSuitDistribution(1, numberOfHearts, 1, 1));
+
+    HandEvaluations threeSuitsWithOneCardHandEvaluations = new HandEvaluations(firstHand);
+    HandEvaluations manyHeartsHandEvaluations = new HandEvaluations(secondHand);
+
+    assertEquals(0, emptyHandEvaluations.getLongestSuitLength());
+    assertEquals(1, threeSuitsWithOneCardHandEvaluations.getLongestSuitLength());
+    assertEquals(numberOfHearts, manyHeartsHandEvaluations.getLongestSuitLength());
+  }
+
+  @Test
+  public void shouldReturnTheCorrectNumberOfDoubletons() {
+
+    Hand firstHand = this.createMockedHandWithDistribution(this.createSuitDistribution(0, 0, 0, 2));
+    Hand secondHand = this.createMockedHandWithDistribution(this.createSuitDistribution(2, 2, 0, 2));
+
+    HandEvaluations oneDoubletonHandEvaluations = new HandEvaluations(firstHand);
+    HandEvaluations threeDoubletonsHandEvaluations = new HandEvaluations(secondHand);
+
+    assertEquals(0, emptyHandEvaluations.getNumberOfDoubletonSuits());
+    assertEquals(1, oneDoubletonHandEvaluations.getNumberOfDoubletonSuits());
+    assertEquals(3, threeDoubletonsHandEvaluations.getNumberOfDoubletonSuits());
+  }
+
+  @Test
+  public void shouldReturnIfItOnlyHasHearts() {
+    when(aceOfSpades.isHeart()).thenReturn(false);
+    when(kingOfHearts.isHeart()).thenReturn(true);
+
+    Hand firstHand = this.createMockedHandWithDistribution(this.createSuitDistribution(0, 1, 0, 0));
+    Hand secondHand = this.createMockedHandWithDistribution(this.createSuitDistribution(1, 1, 0, 0));
+
+    HandEvaluations onlyHeartsHandEvaluations = new HandEvaluations(firstHand);
+    HandEvaluations notOnlyHeartsHandEvaluations = new HandEvaluations(secondHand);
+
+    assertTrue(onlyHeartsHandEvaluations.onlyHasHearts());
+    assertFalse(notOnlyHeartsHandEvaluations.onlyHasHearts());
+  }
+
+  @Test
+  public void shouldReturnIfItHasFiveCardMajorSuit() {
+    Hand firstHand = this.createMockedHandWithDistribution(this.createSuitDistribution(0, 0, 0, 5));
+    Hand secondHand = this.createMockedHandWithDistribution(this.createSuitDistribution(0, 5, 0, 5));
+    Hand thirdHand = this.createMockedHandWithDistribution(this.createSuitDistribution(7, 0, 0, 5));
+
+    HandEvaluations fiveCardClubsHandEvaluations = new HandEvaluations(firstHand);
+    HandEvaluations fiveCardClubsAndHeartsHandEvaluations = new HandEvaluations(secondHand);
+    HandEvaluations fiveCardClubsAndSevenSpadesHandEvaluations = new HandEvaluations(thirdHand);
+
+    assertFalse(emptyHandEvaluations.hasFiveCardMajorSuit());
+    assertFalse(fiveCardClubsHandEvaluations.hasFiveCardMajorSuit());
+    assertTrue(fiveCardClubsAndHeartsHandEvaluations.hasFiveCardMajorSuit());
+    assertTrue(fiveCardClubsAndSevenSpadesHandEvaluations.hasFiveCardMajorSuit());
```

Comment:
I see here... I think the method should be called `hasFiveOrMoreCardsInMajorSuit()`

Reply:
Yes. Renamed it to improve legibility.